projects/cli-files/go-ls/cmd/root.go

```diff
@@ -0,0 +1,51 @@
+package cmd
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+)
+
+func Execute() {
+	help := flag.Bool("h", false, "describes how to use go-ls command")
+	flag.Parse()
+
+	if *help {
+		flag.PrintDefaults()
+		fmt.Println("Usage: go-ls [directories]")
+		return
+	}
+	args := flag.Args()
+
+	if len(args) == 0 {
+		args = []string{"."}
+	}
+
+	for _, y := range args {
+
+		fileInfo, err := os.Stat(y)
+
+		if err != nil {
+			log.Fatal(err)
+		}
+		if fileInfo.IsDir() {
+			listFiles(y)
+		} else {
+			fmt.Printf("%s\n", fileInfo.Name())
+		}
+	}
+
+}
+
+func listFiles(name string) {
+	files, err := os.ReadDir(name)
+
+	if err != nil {
+		log.Fatal(err)
```

Comment:
We try to avoid calling `log.Fatal` (which calls `os.Exit`) outside of the `main` function. Can you restructure this so that this function returns an error in this case, and the main function does the printing and exiting if there was an error?

Some reasons for this:
1. If you ever wanted to test your `listFiles` function, and called it from a test, and it hit an error, it would break your tests and you also wouldn't be able to test the error messages you get in the tests.
2. If you wanted to re-use this function somewhere else, you may _not_ want to `os.Exit` in that case (e.g. if you wanted to retry errors), and it may be problematic for this code to `os.Exit`.
3. In general when looking at functions, it's useful to know whether something can go wrong with it or not. e.g. converting a number to a string _can't_ fail, so a function which does that doesn't return an error - converting a string to a number _can_ fail, so a function which does that may return an error which you need to consider. Whether a function returns an error or not is part of its documentation, and can help you understand how to use it.

Reply:
Ahaa. I remember seeing this a couple of times on the readme. It sticks better now I have made the mistake and fixed it. 
I was also thinking about testing `listFiles `function. Thinking about the behaviour of the function (which was reading the directory path and printing), I realized that I had to limit it to just printing a slice of fileNames, so it does one thing at a time. With the testing, do I have to capture the std output in a variable and assert on it or is there a better way of doing this? I also thought about having it return number of items printed (like I did with `printFileContent`).